provider/dnsimple/dnsimple_test.go

```diff
@@ -215,10 +246,22 @@ func TestNewDnsimpleProvider(t *testing.T) {
 	if err == nil {
 		t.Errorf("Expected to fail new provider on bad token")
 	}
+
 	os.Unsetenv("DNSIMPLE_OAUTH")
+	_, err = NewDnsimpleProvider(endpoint.NewDomainFilter([]string{"example.com"}), provider.NewZoneIDFilter([]string{""}), true)
```

Comment:
This change wasn't really within the scope of this PR but this should be added in any case as without it, `NewDnsimpleProvider` isn't actually tested with an unset `DNSIMPLE_OAUTH` environment variable.